File: Leetcoding-Actions/Explore-Learn/HashTable/happyNumber.java
class Solution {
    public boolean isHappy(int n) {
        Set<Integer> set = new HashSet<>();
        
        while (n > 1) {
            if (set.contains(n)) return false;
            set.add(n);
            int k = 0;
            while (n != 0) {
                int rem = n % 10;
                n = n / 10;
                k += rem * rem;
            }
            n = k;
        }
        return n == 1;
    }
}

File: Leetcoding-Actions/Explore-Learn/HashTable/happyNumber.py
class Solution:
    def isHappy(self, n: int) -> bool:
        nSet = set()
        while n > 1:
            if n in nSet: return False
            nSet.add(n)
            k = 0
            while n:
                n, rem = divmod(n, 10)
                k += rem ** 2
            n = k
        return n == 1
